src/profile/components/profile/__tests__/Profile.test.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { User } from 'oidc-client';
+import { act, cleanup } from '@testing-library/react';
+import { Route, Switch } from 'react-router';
+
+import { getMyProfile } from '../../../../common/test/myProfileMocking';
+import { renderProfileContextWrapper } from '../../../../common/test/componentMocking';
+import { ProfileData } from '../../../../graphql/typings';
+import Profile from '../Profile';
+import {
+  MockedResponse,
+  resetApolloMocks,
+  ResponseProvider,
+} from '../../../../common/test/MockApolloClientProvider';
+import authService from '../../../../auth/authService';
+describe('<Profile />', () => {
+  const RouteOutputAsHTML = () => (
+    <div data-testid="context-as-html">
+      <Switch>
+        <Route path="/login">
+          <div data-testid="location-is-login">/login</div>
+        </Route>
+      </Switch>
+    </div>
+  );
+
+  const renderTestSuite = (responses: MockedResponse[]) => {
+    const responseProvider: ResponseProvider = () =>
+      responses.shift() as MockedResponse;
+    return renderProfileContextWrapper(
+      responseProvider,
+      <React.Fragment>
+        <Profile />
+        <RouteOutputAsHTML />
+      </React.Fragment>
+    );
+  };
+
+  const mockUser = (): void => {
+    const user = ({ profile: { name: 'Mock User' } } as unknown) as User;
+    const userManager = authService.userManager;
+    jest.spyOn(userManager, 'getUser').mockResolvedValueOnce(user);
+  };
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+    cleanup();
+    resetApolloMocks();
+  });
+
+  it('should re-route to /login when user is not authenticated', async () => {
+    const { waitForElement } = await renderTestSuite([]);
+    await waitForElement({ testId: 'location-is-login' });
+  });
+
+  it('should render load indicator and then CreateProfile when profile does not exist', async () => {
```

Comment:
I can't see how this test checks for any load indicator, as the test name suggests.

Reply:
Added missing line.